src/Serialization.cpp

```diff
@@ -1453,17 +1457,35 @@ void Serializer::serialize(const Pipeline &pipeline, const std::string &filename
 
     uint8_t *buf = builder.GetBufferPointer();
     int size = builder.GetSize();
+
+    if (buf != nullptr && size > 0) {
+        result.reserve(size);
+        result.insert(result.begin(), buf, buf + size);
+    } else {
+        user_error << "failed to serialize pipeline!\n";
+    }
+}
+
+void Serializer::serialize(const Pipeline &pipeline, const std::string &filename) {
+    std::vector<uint8_t> data;
```

Comment:
It looks like the Flatbuffer API has limited streaming functionality ... it requires you to always read/write to/from an in-memory data buffer.  But it can tell you how large the buffer should be on a per-object basis, so you could do per object data buffer allocations and read/write to a stream manually.